pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -215,10 +217,18 @@ protected void handleLookup(CommandLookupTopic lookup) {
             return;
         }
 
+        String originalPrincipal = validateOriginalPrincipal(
+                lookup.hasOriginalAuthData() ? lookup.getOriginalAuthData() : null,
+                lookup.hasOriginalAuthMethod() ? lookup.getOriginalAuthMethod() : null,
+                lookup.hasOriginalPrincipal() ? lookup.getOriginalPrincipal() : this.originalPrincipal, requestId,
+                lookup);
+        
+        if (authenticateOriginalAuthData && lookup.hasOriginalAuthData() && originalPrincipal == null) {
+            return;
```

Comment:
this condition seems little tricky. if we fail to validate original principal then `validateOriginalPrincipal()` will send failure response to the client and then thread should not process anything further. but then with this condition there is a possibility to move forward and thread will do further processing if `authenticateOriginalAuthData && lookup.hasOriginalAuthData() =is false`.??